### DISCUSSION TITLE
Problems trying to use the ZMQ setup when deploying the gNB image in Docker

### DISCUSSION THREAD

**Discussion by arosgal** [srsran/srsRAN_Project#877] [General Help]
Hi,

I'm trying to deploy the gNB image of srsRAN using Docker and I want to use the ZMQ-based RF driver. I've followed the installation instructions but I'm running into some issues and I can't find any solution to this specific problem.

First of all, I have followed step by step the tutorial to deploy [srsRAN gNB with srsUE](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html) and it works flawlessly, the thing is that I want to deploy the gNB in docker, same as the 5G core, but with the ZMQ setup.

To use ZMQ, the first step I made is downloading the yaml file [gnb_zmq.yaml](https://docs.srsran.com/projects/project/en/latest/_downloads/a7c34dbfee2b765503a81edd2f02ec22/gnb_zmq.yaml) the tutorial suggests, placed it in the config folder and specified it in the configs section of the docker compose file (I will try to show only the code I modified from the original files, the rest of the code is exactly the same):

```
configs:
  gnb_config.yml:
    file: ${GNB_CONFIG_PATH:-../configs/gnb_zmq.yml} # Path to your desired config file

```

The output if I just do the docker compose like suggested in the README is this:

```
srsran_gnb      | amf was not readable (missing?)
srsran_gnb      | Run with --help for more information.
srsran_gnb      | 
srsran_gnb      | --== srsRAN gNB (commit ) ==--
srsran_gnb      | 
srsran_gnb exited with code 103

```

Right out of the box the default command of the gnb image doesn't work, which is strange because I haven't seen ANY post commenting it, so I tried fixing it removing most of the command to just:

`
command: gnb -c /gnb_config.yml
`

Before trying to deploy it again I added all the [dependencies and cmake args](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html) to the Dockerfile of the gNB image, so the gNB knows that it has to use the ZMQ setup:

```
...

ARG EXTRA_CMAKE_ARGS="-DENABLE_EXPORT=ON -DENABLE_ZEROMQ=ON"

...

# Install srsRAN build dependencies
ARG LIB
RUN apt-get update && \
    apt-get install -y \
    libzmq3-dev \
    cmake \
    make \
    gcc \
    g++ \
    libfftw3-dev \
    libmbedtls-dev \
    libsctp-dev \
    libyaml-cpp-dev \
    libgtest-dev && \
    /src/docker/scripts/install_dependencies.sh build && \
    /src/docker/scripts/install_${LIB}_dependencies.sh build
...
```

I don't know if it is mandatory to install the build tools and dependencies, but I did it just in case. With these modifications to the Dockerfile, I rebuilt the gnb image with the --build flag, but the error this time is:

```
[+] Running 2/0
 ✔ Container open5gs_5gc  Created                                                                   0.0s 
 ✔ Container srsran_gnb   Created                                                                   0.0s 
Attaching to srsran_gnb
srsran_gnb  | [INFO] [UHD] linux; GNU C++ version 13.2.0; Boost_108300; UHD_4.7.0.0-0-unknown
srsran_gnb  | [INFO] [LOGGING] Fastpath logging disabled at runtime.
srsran_gnb  | 
srsran_gnb  | --== srsRAN gNB (commit ) ==--
srsran_gnb  | 
srsran_gnb  | 
srsran_gnb  | The PRACH detector will not meet the performance requirements with the configuration {Format B4, ZCZ 0, SCS 30kHz, Rx ports 1}.
srsran_gnb  | Lower PHY in quad executor mode.
srsran_gnb  | Available radio types: uhd.
srsran_gnb  | Making USRP object with args 'type=b200,num_recv_frames=64,num_send_frames=64'
srsran_gnb  | Failed to open device with address 'type=b200,num_recv_frames=64,num_send_frames=64': LookupError: KeyError: No devices found for ----->
srsran_gnb  | Device Address:
srsran_gnb  |     type: b200
srsran_gnb  |     num_recv_frames: 64
srsran_gnb  |     num_send_frames: 64
srsran_gnb  | 
srsran_gnb  | srsRAN ERROR: Unable to create radio session.

```

Is the ZMQ setup not supposed to work when trying to use the gNB image or it's just I'm doing something wrong?

I want to clarify that I'm not an expert in docker so it's very possible that the problem is superfluous. Sorry in advance if that's the case. I have tried a lot of possible solutions and rebuilt the gNB image countless times, but none of them seem to work.

**Comment by Majdi-Abdelhalim** [comment]
Hi, I want to try the same thing.
Were you able to solve it?
Could you share the Compose file if possible?

Thank you
